exchanges/bitstamp/bitstamp_test.go

```diff
@@ -500,174 +516,165 @@ func TestSubmitOrder(t *testing.T) {
-		t.Errorf("Could not cancel orders: %v", err)
-	case mockTests && err == nil:
-		t.Error("Expecting an error until QA pass is completed")
+	err := b.CancelOrder(context.Background(), &order.Cancel{
+		OrderID: "1453282316578816",
+	})
+	if !mockTests {
+		assert.ErrorContains(t, err, "Order not found")
+	} else {
+		assert.NoError(t, err, "CancelExchangeOrder should not error")
 	}
 }
 
 func TestCancelAllExchangeOrders(t *testing.T) {
 	t.Parallel()
 
 	if !mockTests {
-		sharedtestvalues.SkipTestIfCannotManipulateOrders(t, b, canManipulateRealOrders)
+		sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
 	}
-
 	resp, err := b.CancelAllOrders(context.Background(),
 		&order.Cancel{AssetType: asset.Spot})
-	switch {
-	case !sharedtestvalues.AreAPICredentialsSet(b) && err == nil && !mockTests:
-		t.Error("Expecting an error when no keys are set")
-	case sharedtestvalues.AreAPICredentialsSet(b) && err != nil && !mockTests:
-		t.Errorf("Could not cancel orders: %v", err)
-	case mockTests && err != nil:
-		t.Errorf("Could not cancel orders: %v", err)
-	}
-
+	assert.NoError(t, err, "TestCancelAllExchangeOrders should not error")
 	if len(resp.Status) > 0 {
 		t.Errorf("%v orders failed to cancel", len(resp.Status))
 	}
 }
 
 func TestModifyOrder(t *testing.T) {
 	t.Parallel()
-	sharedtestvalues.SkipTestIfCannotManipulateOrders(t, b, canManipulateRealOrders)
-	_, err := b.ModifyOrder(context.Background(), &order.Modify{AssetType: asset.Spot})
-	if err == nil {
-		t.Error("ModifyOrder() Expected error")
+
+	if !mockTests {
+		sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
 	}
+	_, err := b.ModifyOrder(context.Background(), &order.Modify{AssetType: asset.Spot})
+	assert.ErrorIs(t, err, common.ErrFunctionNotSupported)
 }
 
 func TestWithdraw(t *testing.T) {
 	t.Parallel()
 
 	if !mockTests {
-		sharedtestvalues.SkipTestIfCannotManipulateOrders(t, b, canManipulateRealOrders)
+		sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
 	}
-
-	withdrawCryptoRequest := withdraw.Request{
+	w, err := b.WithdrawCryptocurrencyFunds(context.Background(), &withdraw.Request{
 		Exchange:    b.Name,
 		Amount:      -1,
 		Currency:    currency.BTC,
 		Description: "WITHDRAW IT ALL",
 		Crypto: withdraw.CryptoRequest{
 			Address: core.BitcoinDonationAddress,
 		},
-	}
-
-	_, err := b.WithdrawCryptocurrencyFunds(context.Background(), &withdrawCryptoRequest)
-	switch {
-	case !sharedtestvalues.AreAPICredentialsSet(b) && err == nil && !mockTests:
-		t.Error("Expecting an error when no keys are set")
-	case sharedtestvalues.AreAPICredentialsSet(b) && err != nil && !mockTests:
-		t.Errorf("Withdraw failed to be placed: %v", err)
-	case mockTests && err == nil:
-		t.Error("Expecting an error until QA pass is completed")
+	})
+	if mockTests {
+		assert.NoError(t, err, "WithdrawCryptocurrencyFunds should not error")
+		assert.Equal(t, "1", w.ID, "Withdrawal ID should be correct")
```

Comment:
I receive a panic here:

```
thrasher-corp/gocryptotrader/exchanges/bitstamp/bitstamp_test.go:604
        	Error:      	Received unexpected error:
        	            	amount must be greater than 0
        	Test:       	TestWithdraw
        	Messages:   	WithdrawCryptocurrencyFunds should not error
--- FAIL: TestWithdraw (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x100b30060]

goroutine 66 [running]:
testing.tRunner.func1.2({0x100d2b040, 0x1011ff4b0})
	/usr/local/go/src/testing/testing.go:1545 +0x1c8
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1548 +0x360
panic({0x100d2b040?, 0x1011ff4b0?})
	/usr/local/go/src/runtime/panic.go:914 +0x218
github.com/thrasher-corp/gocryptotrader/exchanges/bitstamp.TestWithdraw(0x0?)
	/Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/bitstamp/bitstamp_test.go:605 +0x150
testing.tRunner(0x14000603a00, 0x100ddad30)
	/usr/local/go/src/testing/testing.go:1595 +0xe8
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:1648 +0x33c
```

You will likely need to verify if `bitstamp.json` matches the expected request/response in mock testing.
You can also change 
```go
assert.NoError(t, err, "WithdrawCryptocurrencyFunds should not error")
```
to
```go
require.NoError(t, err, "WithdrawCryptocurrencyFunds should not error")
```

Reply:
I have changed the assert to require